### PR TITLE
fix panic in ResourceLocation in case of empty pod ip list

### PR DESCRIPTION
**Pull request by khenidak**
**What type of PR is this?**
/kind api-change
 /kind bug
/kind cleanup
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes/kubernetes/commit/54d42e6a658f84ea66593d3107f41db6931a5fbf#r34326319 (cc @liggitt )

**Special notes for your reviewer**:
N/A

**Does this PR introduce a user-facing change?**:
NONE
```release-note
N/A
```

@kubernetes/sig-network-misc

### CODE REVIEW

**Comment by liggitt** [comment]
gofmt error

**Comment by khenidak** [comment]
@liggitt  pushed a fix for the go-fmt issue, let us run all tests, then approve after. thanks

File: pkg/registry/core/pod/strategy.go [hunk 1]
```diff
@@ -239,6 +239,18 @@ func getPod(getter ResourceGetter, ctx context.Context, name string) (*api.Pod,
 	return pod, nil
 }
 
+// returns primary IP for a Pod
```
Comment (bowei) [inline pkg/registry/core/pod/strategy.go]: getPodIP returns the primary IP for a Pod. Returns an empty string if pod is nil or there are no IPs

**Pull request merged**